### PR TITLE
[ENH] Added OneClassSVM to the anomaly detection module

### PR DESCRIPTION
**Pull request by pattplatt**
As discussed on friday, here is the implementation of the OneClassSVM for the Anomaly Detection module. Let me know what you think :)

#### What does this implement/fix? Explain your changes.

This PR includes the implementation of the OneClassSVM from scikit learn for anomaly detection.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/anomaly_detection/_one_class_svm.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    nu : float, default=0.5
+        An upper bound on the fraction of training
+        errors and a lower bound of the fraction of support
+        vectors. Should be in the interval (0, 1]. By default 0.5
+        will be taken.
+
+    shrinking : bool, default=True
+        Whether to use the shrinking heuristic.
+        See the :ref:`User Guide <shrinking_svm>`.
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/_one_class_svm.py:77]: Does this link still work?

File: aeon/anomaly_detection/_one_class_svm.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import numpy as np
+from aeon.anomaly_detection.base import BaseAnomalyDetector
+from aeon.utils.windowing import reverse_windowing, sliding_windows
+
+
+class OneClassSVM(BaseAnomalyDetector):
+    """OneClassSVM for anomaly detection.
+
+    This class implements the OneClassSVM algorithm for anomaly detection
+    fromo sklearn to be used in the aeon framework. All parameters are passed to
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/_one_class_svm.py:18]:
```suggestion
    from sklearn to be used in the aeon framework. All parameters are passed to
```

**Review by SebastianSchmidl — DISMISSED** [review]
Well done 👍🏼 I just have some minor remarks regarding the documentation.
Once the CI passes, we can merge this.

**Comment by SebastianSchmidl** [comment]
Please fix the line-length and run `pre-commit` locally again to address the CI failure in [PR pre-commit](https://github.com/aeon-toolkit/aeon/actions/runs/11803039351/job/32880103397?pr=2342):

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

aeon/anomaly_detection/_one_class_svm.py:33:89: E501 line too long (110 > 88 characters)
```

File: aeon/anomaly_detection/_one_class_svm.py (hunk 1 above)
Comment (pattplatt) [reply aeon/anomaly_detection/_one_class_svm.py]: Thanks for pointing it out, replaced it with the actuall link.

**Review by SebastianSchmidl — APPROVED** [review]

**Pull request merged**